### ISSUE TITLE
A particular type of call to /v2/apps is very slow

### ISSUE THREAD

**Issue by pusherofbrooms** [cloudfoundry/cloud_controller_ng#1314]
Thanks for submitting an issue to `cloud_controller_ng`. We are always trying to improve! To help us, please fill out the following template.

## Issue

In our environment with >4000 applications, api calls which follow the pattern:
GET /v2/apps?q=name%3Asyslogdain&q=space_guid%3A9c331a1b-7311-4666-8cc3-f12f7b35ac80
Are orders of magnitude slower when moving from capi 1.74.0 to capi 1.76.0

## Context

We are using a mysql v5.5.x db.m5.2xlarge db with provisioned iops (1000). Our cf-deployment is currently v7.5.0. We upgraded from cf-deployment v7.1.0. Just after the api instances were upgraded, the slow api call manifested.

## Steps to Reproduce

In an environment as described above, run something like the following:
1) cf space YOUR_SPACE_NAME --guid
# where YOUR_SPACE_NAME is your targeted space
2) cf apps
# and pick an application to inspect
3) cf curl '/v2/apps?q=name%3AYOUR_APP_NAME&q=space_guid%3AYOUR_SPACE_GUID

for fun, on the ccdb, run the following if it's a quiet environment to see the slow query.

select * from information_schema.processlist where COMMAND != 'Sleep' order by TIME;

## Expected result

The api call should return quickly

## Current result

The api call returns in 1 minute 44 seconds

No logs needed. It appears to be a slow query.

## Possible Fix

No suggestions

**Comment by pusherofbrooms** [comment]
As a followup to the slack conversation from yesterday in #capi, we took the downtime in our largest environment and upgraded mysql from 5.5.x to 5.6.x. This improved the problematic api call from about 1m47s to 0.4s.

In our next largest deployment, we haven't done the upgrade and the troublesome api call takes about 12 seconds.

**Comment by ericpromislow** [comment]
I suspect this is due to the addition of the `duplicated_older_process_guids` check, introduced in CAPI 1.76.0, when we allowed more than one process GUID to be associated with a single app GUID, and are filtering out the older duplicates. The query uses a self-join, and it looks like self-joins were sped up going to mysql 5.6.

- Eric, CAPI Community Team

**Comment by pusherofbrooms** [comment]
I must have tested the wrong API call when I declared victory. Upgrading to mysql 5.6 did speed things up a lot compared to mysql 5.5, but in our largest environment, it still takes around 9 seconds to run an api call like:
```
$ time cf curl "/v2/apps?q=name%3Abinding-post&q=space_guid%3A9c331a1b-7311-4666-8cc3-f12f7b35ac80" > /dev/null

real	0m9.173s
user	0m0.117s
sys	0m0.037s
```
The v3 api is much faster for a similar call and returns sub-second
```
$ time cf curl "/v3/apps?names=binding-post&space_guids=9c331a1b-7311-4666-8cc3-f12f7b35ac80" > /dev/null

real	0m0.219s
user	0m0.099s
sys	0m0.035s
```
I still consider this usable, but 9 seconds is a significant fraction of a push and even more of a restart.

We've added about 500 apps in our largest foundation (where this test runs), so there are a total of 4500+ apps there.
